pkg/internal/apis/config/validate_test.go

```diff
@@ -243,3 +243,49 @@ func TestNodeValidate(t *testing.T) {
 		})
 	}
 }
+
+func TestPortValidate(t *testing.T) {
+	cases := []struct {
+		TestName    string
+		Port        int32
+		ExpectError string
+	}{
+		{
+			TestName:    "-1 port",
+			Port:        -1,
+			ExpectError: "",
+		},
+		{
+			TestName:    "valid port",
+			Port:        10,
+			ExpectError: "",
+		},
+		{
+			TestName:    "negative port",
+			Port:        -2,
+			ExpectError: "invalid port number: -2",
+		},
+		{
+			TestName:    "extra port",
+			Port:        65536,
+			ExpectError: "invalid port number: 65536",
+		},
+	}
+
+	for _, tc := range cases {
+		tc := tc //capture loop variable
+		t.Run(tc.TestName, func(t *testing.T) {
+			t.Parallel()
+			err := validatePort(tc.Port)
+			// the error can be:
+			// - nil, in which case we should expect no errors or fail
```

Comment:
nit: this comment seems to cover the cases on the surface, but doesn't actually list them ...?